pinot-core/src/main/java/com/linkedin/pinot/core/startree/DimensionBuffer.java

```diff
@@ -21,51 +21,47 @@
 
 public class DimensionBuffer {
 
-  IntBuffer buffer;
-  private int numDimensions;
+  IntBuffer _intBuffer;
 
   public DimensionBuffer(int numDimensions) {
-    this(numDimensions, ByteBuffer.wrap(new byte[numDimensions * 4]).asIntBuffer());
-    this.numDimensions = numDimensions;
+    this(ByteBuffer.wrap(new byte[numDimensions * Integer.SIZE / 8]).asIntBuffer());
```

Comment:
Seems like you replaced on magic number (4) by another (8). Will using Integer.BYTES solve this?